applications/ConvectionDiffusionApplication/custom_elements/symbolic_qs_convection_diffusion_explicit.cpp

```diff
@@ -0,0 +1,926 @@
+/***********************************************************************************/
+
+template< unsigned int TDim, unsigned int TNumNodes >
+void SymbolicQSConvectionDiffusionExplicit<TDim,TNumNodes>::Calculate(
+    const Variable<double>& rVariable,
+    double& Output,
+    const ProcessInfo& rCurrentProcessInfo)
+{
+    KRATOS_TRY;
+
+    if (rVariable == SCALAR_PROJECTION) {
+        auto& r_geometry = GetGeometry();
+        const unsigned int local_size = r_geometry.size();
+        BoundedVector<double, TNumNodes> rhs_oss;
+        this->CalculateOrthogonalSubgridScaleRHSInternal(rhs_oss,rCurrentProcessInfo);
+        for (unsigned int i_node = 0; i_node < local_size; i_node++) {
+            #pragma omp atomic
+            r_geometry[i_node].GetValue(rVariable) += rhs_oss[i_node];
+        }
+    }
+    else {
+        KRATOS_ERROR << "Variable not implemented to compute OSS projection. Use SCALAR_PROJECTION instead." << std::endl;
+    }
+
+    KRATOS_CATCH("");
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+template< unsigned int TDim, unsigned int TNumNodes >
+void SymbolicQSConvectionDiffusionExplicit<TDim,TNumNodes>::InitializeEulerianElement(
+    ElementData& rData,
+    const ProcessInfo& rCurrentProcessInfo)
+{
+    KRATOS_TRY;
+
+    const ProcessInfo& r_process_info = rCurrentProcessInfo;
+    ConvectionDiffusionSettings::Pointer p_settings = r_process_info[CONVECTION_DIFFUSION_SETTINGS];
+    auto& r_settings = *p_settings;
+
+    // Getting data for the given geometry and integration method
+    const auto& r_geometry = GetGeometry();
+    const unsigned int local_size = r_geometry.size();
+    array_1d<double,TNumNodes> N_aux;
+    GeometryUtils::CalculateGeometryData(r_geometry,rData.DN_DX,N_aux,rData.volume);
+    rData.N_gausspoint = r_geometry.ShapeFunctionsValues(this->GetIntegrationMethod());
+
+    // Initialize some scalar data
+    rData.lumping_factor = 1.00 / double(TNumNodes);
+    rData.diffusivity = 0.0;
+    rData.dynamic_tau = rCurrentProcessInfo[DYNAMIC_TAU];
+
+    for(unsigned int node_element = 0; node_element<local_size; node_element++)
+{
+    // Observations
+    // * SGS time derivative term approximated as (phi-phi_old)/(RK_time_coefficient*delta_time)
+    //   observe that for RK step = 1 ASGS time derivative term = 0 because phi = phi_old (null acceleration wrt step n)
+    // * convective velocity and forcing term:
+    //   RK step 1: evaluated at previous time step
+    //   RK steps 2 and 3: linear interpolation between current and oldprevious time step
+    //   RK step 4: evaluated at current time step
+    // * convective_velocity = velocity - velocity_mesh
+    //   velocity_mesh = 0 in eulerian framework
+    if (r_process_info.GetValue(RUNGE_KUTTA_STEP)==1)
+    {
+        rData.RK_time_coefficient = std::numeric_limits<double>::max();
```

Comment:
Can you comment on this? Shouldn't the RK coefficients be 0.0, 0.5 and 1.0?

Reply:
Such coefficients approximate the derivative
![CodeCogsEqn](https://user-images.githubusercontent.com/43809717/95440954-d7ab5000-0959-11eb-8156-625be599f15a.gif)
which is 0 for RK step 1. What should be zero is `\Delta t` of RK step 1.

Reply:
Yes. This is what I thought. I think this a hacky solution... On top of the fact that your element is only RK compatible... We need to better figure out how to do this.

Reply:
I applied the changes discussed privately.